app/main.py

```diff
@@ -1,25 +1,75 @@
+from __future__ import annotations
+
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: SlideLimitationValidator, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+        self,
+        instance: SlideLimitationValidator,
+        owner: SlideLimitationValidator
+    ) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: SlideLimitationValidator, value: int) -> None:
+        if not (isinstance(value, int)
+                and self.min_amount <= value <= self.max_amount):
```

Comment:
It should just set value. as it's integer range class, values should be integers, thus check it